Need Yum package zlib-devel for container Rocky Linux builds

## Bug Description
<!--A clear and concise description of the problem (Note: A missing feature is not a bug).-->
On my last PR, I boldly assumed that since Rocky Linux also uses Yum and could install all of the packages, the rest of the build would work, but LibMesh fails with the following.
```
configure: error: Can't find or link to the z library. Turn off netCDF-4 and      DAP clients with --disable-netcdf-4 --disable-dap, or see config.log
```

Basically, it turns out that `zlib-devel` needs to be a part of the Yum installs to support Rocky Linux, whereas CentOS has it installed in the bare Docker image `centos:8`.

## Steps to Reproduce
<!--Steps to reproduce the behavior (input file, or modifications to an existing input file, etc.)-->
Attempt a Docker build with `DISTRO=rockylinux` and `DISTRO_VERSION=8.5`.  It will fail at LibMesh configuration.

## Impact
<!--Does this prevent you from getting your work done, or is it more of an annoyance?-->
We are looking to roll out our Rocky Linux image builds soon, so I will be submitting a PR shortly after I'm very sure that this will build.

--- a/docker_ci/yum_installs.sh
+++ b/docker_ci/yum_installs.sh
@@ -60,7 +60,8 @@ yum install -y \
   emacs \
   gtest \
   sudo \
-  file
+  file \
+  zlib-devel
 
 # Clear cache
 yum clean all

